### ISSUE TITLE
The validation probe materializes the full-vocab logits that losses.py exists to avoid

### ISSUE THREAD

**Issue by MatthewLacerda2** [MatthewLacerda2/TinyRefinementModel#208]
**Lane:** partial-cpu — the rewrite and its parity gate are CPU work; the VRAM number wants the card.

## What's wrong

`trm/train/validation.py:32-46` scores the held-out probe with **full logits**, inside the trainer process:

```python
out1 = model(seq1_in, depth=VAL_FIXED_DEPTH, training=False, new_document=True)
out2 = model(seq2_in, depth=VAL_FIXED_DEPTH, training=False, new_document=False)
...
ce = optax.softmax_cross_entropy_with_integer_labels(logits=logits, targets)
```

That is two `[1, 512, 50304]` f32 logit tensors — **~206 MB of transient** — which is precisely the allocation `trm/train/losses.py` exists to eliminate. `chunked_cross_entropy` already solves this and is already used by the grad step; the probe just never adopted it.

## Why it matters more than 206 MB sounds

The probe runs on the trainer's `VAL_EVERY_OPT_STEPS` cadence, into the **same preallocated BFC arena** as training. BFC fragmentation is the documented killer of every base run to date (`docs/findings/2026-08-14-bfc-fragmentation-killed-every-base-run.md`), and `trm/config.py` names the validation probe as part of why the real trainer OOMs where `bench_train_step` does not:

> bench_train_step times a grad step; it never ran the trainer, which also holds the validation probe and the checkpoint managers.

A large, periodic, short-lived allocation is the shape that fragments an arena.

## Fix

Call the model with `training=True` to get pre-head states and score them with `chunked_cross_entropy_rows`, exactly as `trm/train/grad_step.py` does. Keep the probe's semantics otherwise identical: `VAL_ROWS` rows, one row at a time, `VAL_FIXED_DEPTH`, fresh state per row.

## On comparability

This reorders the f32 sum over positions, so recorded val CE shifts in the last decimals (order 1e-6, against a σ≈0.03 noise floor and a ±0.06 time-machine tolerance). **Explicitly accepted — do not design around it.** Nothing pins the value: `tests/expensive/test_golden_run.py` exercises `compute_grad_step` on `UniversalReasoner` and never touches `ValidationProbe`, and `tests/core/test_validation_probe.py` only asserts finite-and-in-range.

## Acceptance

- **CPU, blocking:** chunked and full-logit probe agree to `rtol=1e-5` on a tiny model. Extend `tests/core/test_validation_probe.py`.
- **GPU, follow-up:** peak VRAM across a validation step, before vs after. Report the number.
